### DISCUSSION TITLE
Dummy switch creation

### DISCUSSION THREAD

**Discussion by Jaguaza** [mongoose-os-apps/shelly-homekit#1454] [Ideas]
Proposal:
Create an option in the web interface to create dummy switch.

Objective:
Apart from having the shelly device, having a fake device to be able to add it to automations.

Example:
Automations by a geolocation trigger for connection and disconnection of the alarm in homekit ask for authorization to carry it out in a push notification.

If you create a rule that says when I leave home, turn off and turn on the dumy switch.
And a second rule that says when the switch turns on it turns on the alarm.

Since the switch is not a homekit security device, it does not ask for geolocation authorization.
And since the switch is supposed to be turned on you do not ask for authorization to connect the alarm.

These dummy switche devices are widely used in Homekit.

**Comment by rojer** [comment]
it would actually be easy to do but i'll need more people to confirm its usefulness.

**Comment by andyblac** [comment]
why not just use Shortcuts, to enable / disable the alarm ?

This is what i do, and it does not require confirmation of master user.

![Screenshot 2020-05-17 at 07 57 04](https://user-images.githubusercontent.com/77936/82137915-0d696400-9814-11ea-8554-409ac314b4c6.jpg)
![Screenshot 2020-05-17 at 07 57 13](https://user-images.githubusercontent.com/77936/82137916-10fceb00-9814-11ea-9b68-e1310f1f5f01.jpg)

**Comment by Jaguaza** [comment]
> en realidad sería fácil de hacer, pero necesitaré más personas para confirmar su utilidad.

Ok I don't think that's a problem.

It is a widely used device.

**Comment by Jaguaza** [comment]
> ¿Por qué no usar atajos para activar / desactivar la alarma?
> 
> Esto es lo que hago, y no requiere confirmación del usuario maestro.
> 
> ![Captura de pantalla 2020-05-17 a las 07 57 04](https://user-images.githubusercontent.com/77936/82137915-0d696400-9814-11ea-8554-409ac314b4c6.jpg)
> ![Captura de pantalla 2020-05-17 a las 07 57 13](https://user-images.githubusercontent.com/77936/82137916-10fceb00-9814-11ea-9b68-e1310f1f5f01.jpg)

Good Morning.

It was just a simple usage example.

**Comment by andyblac** [comment]
ah ok...

**Comment by xuleto** [comment]
Estaría genial que implementaseis esta opción a parte de utilizarlo para el sistema de seguridad se podría utilizar en multitud de situaciones. Estaremos atentos!

Translated:
> It would be great if you implemented this option, apart from using it for the security system, it could be used in many situations. We'll be alert!

**Comment by divemasterjm** [comment]
Would be a nice enhancement +1

**Comment by freddair** [comment]
+1

**Comment by edubox** [comment]
+1 I think that this will be useful. Not only in the case that @Jaguaza mentions, but other ways. Many systems have it.

**Comment by jjulveval** [comment]
+1.  It is a great idea and very useful

**Comment by muval** [comment]
Good morning, the updates you are putting in the firm for the shellys are very good, it would be interesting the option of having a dummy switch also when possible.

**Comment by jesegura** [comment]
+1. I join the partner's proposal, I find it interesting to have that dummy switch

**Comment by danimart1991** [comment]
This would be perfect por HomeKit users that look for easy options as possible.

**Comment by Tomer51m** [comment]
+1

**Comment by Borja90** [comment]
+1 
It would be great to have this implemented giving lots of posibilities. Thanks!

**Comment by ElBielas** [comment]
Hi guys I think it's a great idea the dummy option !!!

**Comment by alcar21** [comment]
+1 
i too am very interested

**Comment by rojer** [comment]
wow, wow, ok, i get it :)

**Comment by rojer** [comment]
so, the plan is to add more ["sw" sections](https://github.com/mongoose-os-apps/shelly-homekit/blob/master/mos.yml#L20). currently we only have one for each physical switch but we can have more. set gpio for them to -1 and modify code to ignore all the gpio activity but behave the same in all other respects (state changes, notifications).
we'll need to figure out the UI aspect as well, so we don't populate like 10 of them but allow adding and removing "virtual" switches as needed.

**Comment by Jaguaza** [comment]
> entonces, el plan es agregar más [secciones "sw"](https://github.com/mongoose-os-apps/shelly-homekit/blob/master/mos.yml#L20) . Actualmente solo tenemos uno para cada interruptor físico, pero podemos tener más. establezca gpio para ellos en -1 y modifique el código para ignorar toda la actividad de gpio pero se comporte igual en todos los demás aspectos (cambios de estado, notificaciones).
> también tendremos que descubrir el aspecto de la interfaz de usuario, de modo que no rellenemos como 10 de ellos, sino que permitamos agregar y quitar conmutadores "virtuales" según sea necesario.

thanks

**Comment by alfons1000** [comment]
I’m very interested!!

**Comment by Pixel-Chris** [comment]
I think the optimal solution would be to integrate this to issue #9 !!!

**Comment by Jaguaza** [comment]
> I think the optimal solution would be to integrate this to issue #9 !!!

+1

**Comment by andyblac** [comment]
I guess this should be another easy one, now that code rewrite is done.

**Comment by rojer** [comment]
indeed. but only after the roller shutter is done!

**Comment by andyblac** [comment]
as stateless switch is now coded, this should be fairly easy to add ? could it be added for 2.7.0 ?

**Comment by rojer** [comment]
not for 2.7.0, but it is much easier after recent refactoring effort. maybe soon.
